xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/javascript.vm

```diff
@@ -218,7 +218,13 @@ XWiki.editor = "$!{escapetool.javascript($editor)}";
 XWiki.viewer = "$!{escapetool.javascript($viewer)}";
 XWiki.contextaction = "${xcontext.getAction()}";
 XWiki.skin = '$escapetool.javascript($xwiki.skin)';
+## Special case for when the request is about a resource reference not related to a document.
+## Note that we don't need to do it for the other variables since they are all of type string.
+#if ("$!doc" != "")
 XWiki.docisnew = $!doc.isNew();
+#else
+XWiki.docisnew = false;
```

Comment:
What's the relational for choosing false (which means current document exists) instead of true ?

Reply:
The rationale was to say that if there's no document, then it cannot be new. Note that the fact there's no document is clarified by the fact that all the other values (such as XWiki.currentSpace and XWiki.currentPage) are just empty.

Now I do think that this should be better handled in the future by skipping entirely all those variables and not setting them in such case, if that's properly supported in our scripts which I'm really not sure of right now.

Reply:
bq. The rationale was to say that if there's no document, then it cannot be new.

It cannot really exist either :)

Reply:
I'm not I agree with you on this one. It's possible that there's more `if (!XWiki.docisnew) ... display something` than the contrary but I'm pretty sure the contrary also exists. Moreover semantically speaking it would be really wrong to have a `docisnew` set to true in that case, and probably a bit WTF in case of bug related to it. So I'd be really -1 for it.

Reply:
> semantically speaking it would be really wrong to have a docisnew set to true

I really don't see how indicating that the current document exist while there is no such thing as a current document is semantically correct. None of true or false is semantically correct.

Reply:
> how indicating that the current document exist while there is no such thing as a current document is semantically correct

Well you infer that the document exist because it's not new, while I'm saying that it's not new because it doesn't exist :) But yeah I agree it's a bit playing with words there. I would have preferred to just skip the else part on this if, but AFAIR it's causing some problems. Maybe I could use a null value here, wdyt @mflorea ?